### PR TITLE
Support reading DataField and JsonField value from a file

### PR DESCRIPTION
**Pull request by ducaale**
Ticks one more item from #4

### CODE REVIEW

**Comment by ducaale** [comment]
I need to update `man.roff` to mention the new seperators i.e `=@` and `:=@`

File: src/request_items.rs [hunk 1]
```diff
@@ -210,10 +214,17 @@ impl RequestItems {
                 RequestItem::JsonField(key, value) => {
                     body.insert(key, value);
                 }
+                RequestItem::JsonFieldFromFile(key, value)
+                | RequestItem::DataFieldFromFile(key, value) => {
+                    let mut file = File::open(value)?;
+                    let mut contents = String::new();
+                    file.read_to_string(&mut contents)?;
```
Comment (blyxxyz) [inline src/request_items.rs:221]: You can use [`std::fs::read_to_string`](https://doc.rust-lang.org/std/fs/fn.read_to_string.html) (here and elsewhere)

File: src/request_items.rs [hunk 2]
```diff
@@ -210,10 +214,17 @@ impl RequestItems {
...
+                    body.insert(key, serde_json::from_str(&contents)?);
```
Comment (blyxxyz) [inline src/request_items.rs:222]: I think this is wrong, if it's a `DataFieldFromFile` then it should be inserted as a string, not decoded as JSON

File: src/request_items.rs [hunk 3]
```diff
@@ -228,10 +239,16 @@ impl RequestItems {
...
                 }
                 RequestItem::DataField(key, value) => text_fields.push((key, value)),
+                RequestItem::DataFieldFromFile(key, value) => {
+                    let mut file = File::open(value)?;
+                    let mut contents = String::new();
+                    file.read_to_string(&mut contents)?;
+                    text_fields.push((key, contents));
+                }
                 RequestItem::FormFile(..) => unreachable!(),
                 _ => {}
```
Comment (blyxxyz) [inline src/request_items.rs:253]: I think it's better to not use wildcards for request items anywhere in the codebase any more, so we know we didn't forget anything.

[`to_curl.rs`](https://github.com/ducaale/xh/blob/90dbf65333746845d3c5f21f536e9386ef7e40aa/src/to_curl.rs#L271) is wrong right now. There might be others.

**Review by blyxxyz — CHANGES_REQUESTED** [review]
Looks good overall. Maybe there should be a few integration tests?

File: src/to_curl.rs [hunk 4]
```diff
@@ -253,13 +254,17 @@ pub fn translate(args: Cli) -> Result<Command> {
...
+                RequestItem::JsonField(..) | RequestItem::JsonFieldFromFile(..) => {
                     return Err(anyhow!("JSON values are not supported in multipart fields"));
                 }
                 RequestItem::DataField(key, value) => {
                     cmd.flag("-F", "--form");
                     cmd.push(format!("{}={}", key, value));
                 }
+                RequestItem::DataFieldFromFile(key, value) => {
+                    cmd.flag("-F", "--form");
+                    cmd.push(format!("{}={}", key, fs::read_to_string(value)?));
```
Comment (blyxxyz) [inline src/to_curl.rs:266]:
```suggestion
                    cmd.push(format!("{}=<{}", key, value));
```

**Review by blyxxyz — APPROVED** [review]
LGTM

**Pull request merged**